PSA tools docs

orenc17 opened the pull request:
### Description

Create documentation for all the PSA python tools & scripts.

This PR depends on:
* #9894 - merged
* #9918 - merged
* #9928 - merged

Do not run CI untill all of the above are merged
<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [X] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers
<!--
    Optional
    Request additional reviewers with @username
-->

@mikisch81 @ARMmbed/mbed-os-maintainers 
### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

Commit:
Add README.md for tools/psa

theotherjimmy commented on tools/psa/README.md:
````diff
@@ -0,0 +1,50 @@
+
+`release.py` is the script assigned with compiling the secure images.
+
+```python
````
I don't think this code block contains python

orenc17 replied:
thanks, removed that

Commit:
Remove python from code block

cmonr commented on tools/psa/README.md:
```diff
@@ -0,0 +1,50 @@
+each implementation requires a set of auto generated files describing the secure partitions.
+
+* `generate_partition_code.py` - Generate files for both implementations.
+* `generate_mbed_spm_partition_code.py` - Generate files for BMED_SPM.
```
I'd like to know what `BMED_SPM` is 😁

orenc17 replied:
![2vecn2](https://user-images.githubusercontent.com/17570086/53920197-04f49580-4075-11e9-82e4-c21c85471859.jpg)

cmonr commented on tools/psa/README.md:
```diff
@@ -0,0 +1,50 @@
+each implementation requires a set of auto generated files describing the secure partitions.
+
+* `generate_partition_code.py` - Generate files for both implementations.
+* `generate_mbed_spm_partition_code.py` - Generate files for BMED_SPM.
```
A question about the script name.

Is there a reason it's named `generate_mbed_spm_partition_code.py` instead of `generate_spm_partition_code.py`?

orenc17 replied:
Branding

cmonr reviewed (changes requested):
Typo needs fixing.

Commit:
Fix typo

Commit:
Edit README.md

Edit file for minor grammar and style tweaks, mostly for consistent capitalization and tense across docs.
